mlflow/store/artifact/utils/models.py

```diff
@@ -32,39 +37,59 @@ def _get_latest_model_version(client, name, stage):
     return max(int(x.version) for x in latest)
 
 
+class ParsedModelUri(NamedTuple):
+    name: str
+    version: Optional[str] = None
+    stage: Optional[str] = None
+    alias: Optional[str] = None
+
+
 def _parse_model_uri(uri):
     """
-    Returns (name, version, stage). Since a models:/ URI can only have one of
-    {version, stage, 'latest'}, it will return
-        - (name, version, None) to look for a specific version,
-        - (name, None, stage) to look for the latest version of a stage,
-        - (name, None, None) to look for the latest of all versions.
+    Returns a ParsedModelUri tuple. Since a models:/ URI can only have one of
+    {version, stage, 'latest', alias}, it will return
+        - (name, version, None, None) to look for a specific version,
+        - (name, None, stage, None) to look for the latest version of a stage,
+        - (name, None, None, None) to look for the latest of all versions.
+        - (name, None, None, alias) to look for a registered model alias.
     """
     parsed = urllib.parse.urlparse(uri)
     if parsed.scheme != "models":
         raise MlflowException(_improper_model_uri_msg(uri))
-
     path = parsed.path
-    if not path.startswith("/") or len(path) <= 1:
+    r = re.compile(r"^\/(?P<model_name>[\w \-]+)(?P<suffix>\/[\w]+)?(?P<alias>@[\w\-]+)?$")
+    m = r.match(path)
+    if m is None:
         raise MlflowException(_improper_model_uri_msg(uri))
-    parts = path[1:].split("/")
-
-    if len(parts) != 2 or parts[0].strip() == "":
+    model_name = m.group("model_name") if "model_name" in m.groupdict() else None
+    suffix = m.group("suffix") if "suffix" in m.groupdict() else None
+    alias = m.group("alias") if "alias" in m.groupdict() else None
+    if (
+        (model_name is None or model_name == "")
```

Comment:
```suggestion
```

I think this line can be removed because the `model_name` group is not optional.

Reply:
Yeah I replaced it with `model_name.strip()` to make sure the name isn't just whitespace